Flaky TestWorkerRestart.testRestartDuringQuery

```
Error:    TestWorkerRestart.testRestartDuringQuery:101 
Expecting message:
  "Could not communicate with the remote task. The node may have crashed or be under too much load. This is probably a transient issue, so please retry your query in a few minutes. (127.0.0.1:36561)"
to be found for regex:
  "^Expected response code from \S+ to be 200, but was 500|Error fetching \S+: Expected response code to be 200, but was 500"
but did not.

Throwable that failed the check:

io.trino.testing.QueryFailedException: Could not communicate with the remote task. The node may have crashed or be under too much load. This is probably a transient issue, so please retry your query in a few minutes. (127.0.0.1:36561)
	at io.trino.testing.AbstractTestingTrinoClient.execute(AbstractTestingTrinoClient.java:133)
	at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:558)
	at io.trino.testing.DistributedQueryRunner.execute(DistributedQueryRunner.java:541)
	at io.trino.testing.QueryRunner.execute(QueryRunner.java:82)
	at io.trino.tests.TestWorkerRestart.lambda$testRestartDuringQuery$1(TestWorkerRestart.java:96)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:317)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
	Suppressed: java.lang.Exception: SQL: SELECT count(*) FROM tpch.sf1000000000.lineitem -- 31cfdfda-e29f-4962-a423-3bf23e66fbf4
		at io.trino.testing.DistributedQueryRunner.executeInternal(DistributedQueryRunner.java:565)
		... 7 more
Caused by: io.trino.spi.TrinoException: Could not communicate with the remote task. The node may have crashed or be under too much load. This is probably a transient issue, so please retry your query in a few minutes. (127.0.0.1:36561)
	at io.trino.server.remotetask.ContinuousTaskStatusFetcher.updateTaskStatus(ContinuousTaskStatusFetcher.java:253)
	at io.trino.server.remotetask.ContinuousTaskStatusFetcher$TaskStatusResponseCallback.success(ContinuousTaskStatusFetcher.java:176)
	at io.trino.server.remotetask.ContinuousTaskStatusFetcher$TaskStatusResponseCallback.success(ContinuousTaskStatusFetcher.java:166)
	at io.trino.server.remotetask.SimpleHttpResponseHandler.onSuccess(SimpleHttpResponseHandler.java:49)
	at io.trino.server.remotetask.SimpleHttpResponseHandler.onSuccess(SimpleHttpResponseHandler.java:27)
	at com.google.common.util.concurrent.Futures$CallbackListener.run(Futures.java:1137)
	at io.airlift.concurrent.BoundedExecutor.drainQueue(BoundedExecutor.java:79)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1144)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:642)
	at java.base/java.lang.Thread.run(Thread.java:1570)
```
https://github.com/trinodb/trino/actions/runs/9670080345/job/26678113228?pr=22499

--- a/testing/trino-tests/src/test/java/io/trino/tests/TestWorkerRestart.java
+++ b/testing/trino-tests/src/test/java/io/trino/tests/TestWorkerRestart.java
@@ -100,7 +100,8 @@ public class TestWorkerRestart
                         .isInstanceOf(ExecutionException.class)
                         .cause().hasMessageFindingMatch("^Expected response code from \\S+ to be 200, but was 500" +
                                                         "|Error fetching \\S+: Expected response code to be 200, but was 500" +
-                                                        "|Could not communicate with the remote task. The node may have crashed or be under too much load");
+                                                        "|Could not communicate with the remote task. The node may have crashed or be under too much load" +
+                                                        "|Error fetching \\S+: Content-Type header is not set");
 
                 // Ensure that the restarted worker is able to serve queries.
                 assertThat((long) queryRunner.execute("SELECT count(*) FROM tpch.tiny.lineitem").getOnlyValue())

